packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -4644,6 +4670,7 @@ class EditableTextState extends State<EditableText>
       TargetPlatform.macOS || TargetPlatform.linux || TargetPlatform.windows => true,
     };
     final bool shouldSelectAll =
+        widget.selectAllOnFocus &&
         widget.selectionEnabled &&
         (kIsWeb || isDesktop) &&
```

Comment:
I think we can get rid of this line. This logic is encoded in defaultSelectAllOnFocus. Furthermore, if we remove this line, then selectAllOnFocus applies equally to all platforms, which I think is more clear for developers that are trying to understand what this parameter does.